### PR TITLE
Update EXTCODEHASH to handle reverts

### PR DESCRIPTION
**Pull request by z2trillion**
Update specs for EXTCODEHASH to match its implementation in https://github.com/appliedzkp/zkevm-circuits/pull/353

### CODE REVIEW

**Comment by z2trillion** [comment]
@ed255

**Review by ed255 — APPROVED** [review]
Thanks for the update!  LGTM!

File: tests/evm/test_extcodehash.py [hunk 1]
```diff
@@ -48,11 +65,24 @@ def test_extcodehash(address: U160, nonce: U256, balance: U256, code: bytes, is_
     tx_id = 1
     call_id = 1
 
+    rw_counter_end_of_reversion = 0 if is_persistent else 9
+    state_write_counter = 0
+
     rw_table = set(
         RWDictionary(0)
```
Comment (icemelon) [inline tests/evm/test_extcodehash.py:72]:
```suggestion
        RWDictionary(1)
```
rw counter should start with 1

**Review by icemelon — APPROVED** [review]
otherwise LGTM.

**Pull request merged**